demos/python_demos/bert_question_answering_demo/bert_question_answering_demo.py

```diff
@@ -0,0 +1,370 @@
+    args = build_argparser().parse_args()
+
+    if args.colors:
+        COLOR_RED = "\033[91m"
+        COLOR_RESET = "\033[0m"
+    else:
+        COLOR_RED = ""
+        COLOR_RESET = ""
+
+    # load vocabulary file for model
+    log.info("Loading vocab file:\t{}".format(args.vocab))
+    with open(args.vocab, "r", encoding="utf-8") as r:
+        vocab = {t.rstrip("\n"): i for i, t in enumerate(r.readlines())}
+    log.info("{} tokens loaded".format(len(vocab)))
+
+    # get context as a string (as we might need it's length for the sequence reshape)
+    context = get_context(args.input)
+    log.info("Size: {} chars".format(len(context)))
+    log.info("Context: " + COLOR_RED + context + COLOR_RESET)
+    # encode context into token ids list
+    c_tokens_id, c_tokens_se = text_to_tokens(context, vocab)
+
+    log.info("Initializing Inference Engine")
+    ie = IECore()
+    version = ie.get_versions(args.device)[args.device]
+    version_str = "{}.{}.{}".format(version.major, version.minor, version.build_number)
+    log.info("Plugin version is {}".format(version_str))
+
+    # read IR
+    model_xml = args.model
+    model_bin = os.path.splitext(model_xml)[0] + ".bin"
+    log.info("Loading network files:\n\t{}\n\t{}".format(model_xml, model_bin))
+    ie_encoder = ie.read_network(model=model_xml, weights=model_bin)
+
+    if args.reshape:
+        # reshape the sequence length to the context + maximum question length (in tokens)
+        first_input_layer = next(iter(ie_encoder.inputs))
+        c = ie_encoder.inputs[first_input_layer].shape[1]
+        # find the closest multiple of 64, if it is smaller than current network's sequence length, let' use that
+        seq = min(c, np.ceil((len(c_tokens_id) + args.max_question_token_num) / 64) * 64)
+        if seq < c:
+            input_info = list(ie_encoder.inputs)
+            new_shapes = dict([])
+            for i in input_info:
+                n, c = ie_encoder.inputs[i].shape
+                new_shapes[i] = [n, seq]
+                log.info("Reshaped input {} from {} to the {}".format(i, ie_encoder.inputs[i].shape, new_shapes[i]))
+            log.info("Attempting to reshape the network to the modified inputs...")
+            try:
+                ie_encoder.reshape(new_shapes)
+                log.info("Successful!")
+            except RuntimeError:
+                log.error("Failed to reshape the network, please retry the demo without '-r' option")
+                sys.exit(-1)
+        else:
+            log.info("Skipping network reshaping,"
+                     " as (context length + max question length) exceeds the current (input) network sequence length")
+
+    # check input and output names
+    input_names_model = list(ie_encoder.inputs.keys())
+    output_names_model = list(ie_encoder.outputs.keys())
+    input_names = eval(args.input_names)
+    output_names = eval(args.output_names)
```

Comment:
It seems unnecessary to enable arbitrary code execution for these arguments, when all you need is a list of names. It's also inconvenient for the user to enter Python list literals as arguments. I would suggest to just split the arguments on commas.

Reply:
fixed